### PR TITLE
Fixing memory leak for expansion-panel (#8403)

### PR DESCRIPTION
**Pull request by AnthonNilsson**

### CODE REVIEW

**Comment by Emanjonil** [comment]
I signed it!

File: src/lib/expansion/expansion-panel.ts [hunk 1]
```diff
@@ -142,6 +142,7 @@ export class MatExpansionPanel extends _MatExpansionPanelMixinBase
   }
 
   ngOnDestroy() {
+    super.ngOnDestroy();
```
Comment (crisbeto) [inline src/lib/expansion/expansion-panel.ts]: This used to be necessary in Angular 2.x, but it should be called automatically in the newer versions.

Comment (AnthonNilsson) [reply src/lib/expansion/expansion-panel.ts]: Thanks for the response!

But this does not seem to be the case here, not in my SPA application at least. I am using angular 5.0.1 and material2 5.0.0-rc0 together with angular cli 1.5.0.
And if checking my answer in (#8403) this problem also seems to be existing in the angular material homepage.

I get a Detatched DOM tree for each expansion panel I have disposed. The ngOnDestroy is never called in CdkAccordionItem.

If manually changing in the material2 js generated code, for MatExpansionPanel, in my node_modules folder expansion.js
image

or for es5 expansion.es5.js:
image

Then the ngOnDestroy is called in CdkAccordionItem and the issue with Detatched DOM trees are removed.

**Comment by AnthonNilsson** [comment]
I am very sorry but after failing to correct the Author information to my correct email I have now made a new pull request which contains my correct email address for Author. (#8410)

**Pull request closed without merging**